cpp/dolfinx/io/adios2_utils.h

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2021 Jørgen S. Dokken
+//
+// This file is part of DOLFINx (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+#pragma once
+
+#ifdef HAS_ADIOS2
+#include "utils.h"
+#include <adios2.h>
+#include <string>
+#include <vector>
+#include <xtensor/xtensor.hpp>
+#include <xtensor/xview.hpp>
+
+namespace dolfinx
+{
+namespace mesh
+{
+class Mesh;
+}
+
+namespace fem
+{
+template <typename T>
+class Function;
+}
+
+namespace io
+{
+// Convenience functions for declaring attributes and variables with adios2
+namespace adios2_utils
+{
+
+//-----------------------------------------------------------------------------
+/// Extract the mesh connectivity for all cells of a given mesh
```

Comment:
I don't understand what's returned here, and it doesn't look to be adios2 specific.

Reply:
It is extracting the mesh connectivity for each cell. It is not the topology. i.e. for two triangles you could get
[[0, 1, 2], [1,2,3]]

Reply:
I don't see how that's not the cell topology.

Reply:
The DOLFINx topology is first order.
The connectivity here supports higher order cells. 
We could try to extend https://github.com/FEniCS/dolfinx/blob/d316ef77c58883f09d0e68486aa1caa24a10f6ec/cpp/dolfinx/mesh/utils.h instead, or make a similar function in utils.